### PR TITLE
Vega fallback loader

### PR DESCRIPTION
**Pull request by fiskus**
If a request with "Range: bytes" fails then request full file contents.

### CODE REVIEW

File: catalog/app/components/Preview/loaders/Json.js [hunk 1]
```diff
@@ -152,14 +153,41 @@ function JsonLoader({ gated, handle, children }) {
...
     Ok: ({ firstBytes, contentLength }) =>
       detectSchema(firstBytes) ? (
         <VegaLoader {...{ handle, children, gated: contentLength > MAX_SIZE }} />
       ) : (
         <JsonLoader {...{ handle, children, gated: contentLength > MAX_SIZE }} />
       ),
+
+    Err: (err) => {
+      if (!error) {
+        sentry('captureException', error)
```
Comment (nl0) [inline catalog/app/components/Preview/loaders/Json.js:185]: did you mean this?
```suggestion
        sentry('captureException', err)
```

**Review by nl0 — COMMENTED** [review]
seems like this code will try to load the whole file regardless of its size, and i'm afraid this can be a problem for like 10MB+ files

**Comment by fiskus** [comment]
I added HEAD request to check the file's size

**Comment by fiskus** [comment]
We discussed and decided to move head request BEFORE bytes request, because it makes more sense to make a correct request instead of making an error request and do something to fix it after.

**Review by nl0 — CHANGES_REQUESTED** [review]
as per zoom discussion: we should use head request before scanning the first bytes

**Comment by fiskus** [comment]
I moved `HEAD` request before `Range: bytes` request

**Review by nl0 — APPROVED** [review]

**Pull request merged**